### PR TITLE
[Cherry-pick] Fixes #721 - Added support for Tailoring file entity

### LINKED ISSUE #721: Support for Tailoring file entity is missing in Nailgun.

### PR DESCRIPTION
**Pull request by jameerpathan111**
Cherry-pick PR: https://github.com/SatelliteQE/nailgun/pull/722

### CODE REVIEW

File: nailgun/entities.py [hunk 1]
```diff
@@ -7580,6 +7580,73 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
...
+        EntitySearchMixin,
+        EntityUpdateMixin):
+    """A representation of a Tailoring File entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'name': entity_fields.StringField(
+                required=True,
+                str_type='alpha',
+                length=(4, 30),
```
Comment (jyejare) [inline nailgun/entities.py:7597]: Is it mentioned somewhere about the length limit? I don't think so.

What If I provide name of 3 chars , will that fail ?

File: nailgun/entities.py [hunk 2]
```diff
@@ -7580,6 +7580,73 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
...
+            with open(kwargs['scap_file']) as input_file:
+                kwargs['scap_file'] = input_file.read()
+        self._meta = {
+            'api_path': 'api/v2/compliance/tailoring_files',
+            'server_modes': ('sat')
+        }
+        super(TailoringFile, self).__init__(server_config, **kwargs)
+
+    def create(self, create_missing=None):
+        """Do extra work to fetch a complete set of attributes for this entity.
```
Comment (jyejare) [inline nailgun/entities.py:7616]: And whats that extra work? worth mentioning ?

Comment (jyejare) [reply nailgun/entities.py:7616]: Remove `extra work` with actual work text.

File: nailgun/entities.py [hunk 3]
```diff
@@ -7982,12 +8049,27 @@ def __init__(self, server_config=None, **kwargs):
...
+            with open(kwargs['scap_file']) as input_file:
+                kwargs['scap_file'] = input_file.read()
         self._meta = {
             'api_path': 'api/compliance/scap_contents',
             'server_modes': ('sat'),
         }
         super(ScapContents, self).__init__(server_config, **kwargs)
 
+    def create(self, create_missing=None):
+        """Do extra work to fetch a complete set of attributes for this entity.
```
Comment (jyejare) [inline nailgun/entities.py:8062]: same here `extra work`

**Comment by jameerpathan111** [comment]
@jyejare updated the pr.

File: nailgun/entities.py [hunk 4]
```diff
@@ -7580,6 +7580,72 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
...
+            with open(kwargs['scap_file']) as input_file:
+                kwargs['scap_file'] = input_file.read()
+        self._meta = {
+            'api_path': 'api/v2/compliance/tailoring_files',
+            'server_modes': ('sat')
+        }
+        super(TailoringFile, self).__init__(server_config, **kwargs)
+
+    def create(self, create_missing=None):
+        """Fetch a complete set of attributes like locations, organizations for this entity.
```
Comment (jyejare) [inline nailgun/entities.py:7615]: The override of function `create` is doing something else that just docstringed `here`. You can refer the mentioned bugzilla in that docstring.

Anyways its a non blocking comment,

Comment (jameerpathan111) [reply nailgun/entities.py:7615]: I was facing issue related to not getting full list of attributes and saw most of other entities were facing the same. I have followed same approach as them and they all were referring to same bz.

**Pull request merged**